dask/order.py

```diff
@@ -202,61 +202,77 @@ def dependencies_key(x):
             StrComparable(x),
         )
 
+    def finish_now_key(x):
+        return (-len(dependencies[x]), StrComparable(x))
+
+    keys = {key: val[1] for key, val in metrics.items()}  # min_dependencies
     result = {}
     i = 0
 
-    # Nodes in the current weakly connected subgraph that are not part of the current
-    # DFS along dependencies.  The next DFS will begin from these nodes.  `outer_stack`
-    # is populated from `init_stack` and the dependents of `outer_stack_seeds`.
-    outer_stack = []
-
-    # Nodes get added to this as they complete.  We will populate `outer_stack` from
-    # dependents of these nodes.  We consider nodes using a FIFO policy (yes, we could
-    # have used collections.deque, but a list with the current index is faster).
-    # Note that this does not include nodes with no dependents!
-    outer_stack_seeds = []
-    outer_stack_seeds_index = 0
-
     # Used to perform a DFS along dependencies.  Once emptied (when traversing dependencies),
     # this continues (along dependents) along a path from the initial (leaf) node down to a
     # root node (a tactical goal).  This ensures this root node will get calculated before
     # we consider any other dependents from `outer_stack_seeds`.
-    inner_stack = []
+    inner_stacks = []
+    inner_stacks_append = inner_stacks.append
+    inner_stacks_extend = inner_stacks.extend
+    inner_stacks_pop = inner_stacks.pop
+
+    current_epoch = []
+    current_epoch_extend = current_epoch.extend
+    current_epoch_pop = current_epoch.pop
+    next_epoch = {}
+    next_era = {}  # era > epoch
```

Comment:
A comment somewhere on "eras" vs. "epochs" would be helpful. IIRC on our call today this clicked for me when you tied these to changes in tactical goals.

Reply:
Definitely agree.  I still need to add a lot of code comments.